### PR TITLE
[Issue #987] Replace is_err() with suitable messages

### LINKED ISSUE #987: Remove is_err() from linera-core/src/unit_tests/worker_tests.rs
The goal of this cleanup task is to replace `is_err` by `assert!(matches!(...))` as we did in `linera-core/src/unit_tests/client_tests.rs` already.

### PR DESCRIPTION
**Pull request by ashu26jha**
## Motivation

To fix #987 

## Proposal

Tried replace is_err() with suitable messages with already exisiting messages

## Release Plan

- [ ] All good!
- [ ] Need to bump the major/minor version number in the next release of the crates.
- [ ] Need to update the developer manual.
- [ ] This PR is adding or removing Cargo features.
- [ ] Release is blocked and/or tracked by other issues (see links above)

## Reviewer Checklist

- [ ] The title and the summary of the PR are short and descriptive.
- [ ] The proposed solution achieves the goals stated in the PR.
- [ ] The test plan is reproducible and provides sufficient coverage.
- [ ] The release plan is adequate.
- [ ] The commits correspond to distinct logical changes.
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
- [ ] The proposed changes look correct.
- [ ] The CI is passing.
- [ ] All of the above!

### CODE REVIEW

**Comment by ashu26jha** [comment]
How can I fix these failing test 🙃?

**Comment by ma2bd** [comment]
You need to assert the correct errors :) Try to print them with `.unwrap()`

**Comment by ashu26jha** [comment]
> You need to assert the correct errors :) Try to print them with `.unwrap()`

Can you elaborate on this please?

**Comment by afck** [comment]
For example currently it says:
`'assertion failed: matches!(worker.fully_handle_certificate(certificate, vec! []).await,\n    Err(WorkerError :: ChainError(error)) if matches!\n    (* error, ChainError :: CertificateSignatureVerificationFailed { .. }))', linera-core/src/unit_tests/worker_tests.rs:1297:5`
So we only know that the result doesn't match the pattern. To find out what the result actually is, you could replace this with:
`worker.fully_handle_certificate(certificate, vec! []).await.unwrap();`
temporarily; then the test should also fail, but print the actual error. (Or if it doesn't fail we know that the result was unexpectedly `Ok`!)

Alternatively, you could put `dbg!(…)` around `worker.fully_handle_certificate(certificate, vec! []).await` and run the test with `RUST_LOG=debug`, so it prints it.

Or you could make the `assert!` print the value if it fails:

```rust
let result = worker.fully_handle_certificate(certificate, vec![]).await;
assert!(
    matches!(result, Err(WorkerError::ChainError(error))
        if matches!(*error, ChainError::CertificateSignatureVerificationFailed{..})),
    "Unexpected result: {:?}",
    result
);
```

File: linera-core/src/unit_tests/worker_tests.rs [hunk 1]
```diff
@@ -1282,10 +1306,10 @@ where
         None,
     )
     .await;
-    assert!(worker
+    assert!(matches!(worker
         .fully_handle_certificate(certificate, vec![])
-        .await
-        .is_err());
+        .await,
+        Err(WorkerError::ChainError(error)) if matches!(*error, ChainError::InactiveChain{..})));
```
Comment (afck) [inline linera-core/src/unit_tests/worker_tests.rs:1312]: @ma2bd: Is this intended?
It looks like it should use `ChainDescription::Root(2)` above, and fail due to the wrong signature instead?

File: linera-core/src/unit_tests/worker_tests.rs [hunk 2]
```diff
@@ -1085,7 +1106,10 @@ where
     let block_proposal = make_first_block(ChainId::root(1))
         .with_simple_transfer(Recipient::root(2), Amount::from_tokens(1000))
         .into_simple_proposal(&sender_key_pair);
-    assert!(worker.handle_block_proposal(block_proposal).await.is_err());
+    assert!(matches!(
+        worker.handle_block_proposal(block_proposal).await,
+        Err(WorkerError::ChainError(error)) if matches!(*error, ChainError::ExecutionError(ExecutionError::SystemError(SystemExecutionError::InsufficientFunding { .. }), ChainExecutionContext::Operation(_)))
```
Comment (afck) [inline linera-core/src/unit_tests/worker_tests.rs:1111]: Please manually format this; `rustfmt` sometimes gives up. :sweat_smile:

File: linera-core/src/unit_tests/worker_tests.rs [hunk 3]
```diff
@@ -383,10 +386,12 @@ where
...
         .into_simple_proposal(&sender_key_pair);
-    assert!(worker
-        .handle_block_proposal(zero_amount_block_proposal)
-        .await
-        .is_err());
+    assert!(matches!(
+        worker
+            .handle_block_proposal(zero_amount_block_proposal)
+            .await,
+            Err(WorkerError::ChainError(error)) if matches!(*error, ChainError::ExecutionError(ExecutionError::SystemError(SystemExecutionError::IncorrectTransferAmount), ChainExecutionContext::Operation(_)))
```
Comment (afck) [inline linera-core/src/unit_tests/worker_tests.rs:393]: Please format manually; lines should be ≤ 100 characters.

**Review by afck — COMMENTED** [review]
Looks good to me! Just two nit-picks!.

File: linera-core/src/unit_tests/worker_tests.rs (hunk 2 above)
Comment (ashu26jha) [reply linera-core/src/unit_tests/worker_tests.rs:1111]: Sure! These do need beautification

File: linera-core/src/unit_tests/worker_tests.rs (hunk 1 above)
Comment (ma2bd) [reply linera-core/src/unit_tests/worker_tests.rs:1337]: Oops. Happy to fix the test after this PR lands.

**Comment by ashu26jha** [comment]
Apologies for the clumsy work, would be more clinical in the next one!

**Review by ma2bd — APPROVED** [review]

**Comment by ma2bd** [comment]
Thanks @ashu26jha for the PR!

**Pull request merged**